app/ide-desktop/lib/dashboard/src/authentication/src/authentication/providers/session.tsx

```diff
@@ -26,18 +27,52 @@ const SessionContext = react.createContext<SessionContextType>(
 // =======================
 
 interface SessionProviderProps {
+  /** URL that the content of the app is served at, by Electron.
+   *
+   * This **must** be the actual page that the content is served at, otherwise the OAuth flow will
+   * not work and will redirect the user to a blank page. If this is the correct URL, no redirect
+   * will occur (which is the desired behaviour).
+   *
+   * The URL includes a scheme, hostname, and port (e.g., `http://localhost:8080`). The port is not
+   * known ahead of time, since the content may be served on any free port. Thus, the URL is
+   * obtained by reading the window location at the time that authentication is instantiated. This
+   * is guaranteed to be the correct location, since authentication is instantiated when the content
+   * is initially served. */
+  mainPageUrl: URL;
+  registerAuthEventListener: listen.ListenFunction;
   userSession: () => Promise<results.Option<cognito.UserSession>>;
   children: react.ReactNode;
 }
 
-// eslint-disable-next-line @typescript-eslint/naming-convention
 export function SessionProvider(props: SessionProviderProps) {
-  const { children, userSession } = props;
+  const { mainPageUrl, children, userSession, registerAuthEventListener } =
+    props;
 
   /** Flag used to avoid rendering child components until we've fetched the user's session at least
    * once. Avoids flash of the login screen when the user is already logged in. */
   const [initialized, setInitialized] = react.useState(false);
 
+  /** Produces a new object every time.
```

Comment:
looking at it again, line 55-75 is a *lot* of docs for something that should be simple. i don't really have any ideas though

Reply:
In this case, this is such a magic (at least to me) that I like that we have docs for it.